File: cpp/ex2_1_2.cpp
#include <string>

using namespace std;

class EmailDst {
public:
    virtual void sendEmail(string content)=0;
};

class Person : public EmailDst {
public:
    virtual void sendEmail(string content) {}
};

void sendNewYearEmail(EmailDst *dst) {
    dst->sendEmail("");
}

int main() {
    Person *pr = new Person();
    sendNewYearEmail(pr);
}

File: kotlin/ex2_1_2.kt
abstract class EmailDst {
    abstract fun sendEmail(content: String): Unit
}

class Person : EmailDst() {
    override fun sendEmail(content: String): Unit {}
}

fun sendNewYearEmail(dst: EmailDst): Unit {
    dst.sendEmail("")
}

fun main() {
    val pr: Person = Person()
    sendNewYearEmail(pr)
}
